File: src/main/java/org/dbshell/reflection/utils/DatabaseMetadataUtil.java
package org.dbshell.reflection.utils;

import java.lang.reflect.Method;
import java.lang.reflect.Modifier;
import java.sql.DatabaseMetaData;
import java.util.Arrays;
import java.util.Map;
import java.util.stream.Collectors;

public class DatabaseMetadataUtil {

    public static Map<String, Method> getPrimitivesFromDBMetadata(DatabaseMetaData dbmd) {
        Class klazz = dbmd.getClass();
        return Arrays.stream(klazz.getMethods())
                .filter(m -> {
                    Integer modifier = m.getModifiers();
                    return m.getParameterCount() == 0 &&
                            Modifier.isPublic(modifier) &&
                            (m.getReturnType().equals(String.class) ||
                                    m.getReturnType().equals(Integer.TYPE) ||
                                    m.getReturnType().equals(Boolean.TYPE) ||
                                    m.getReturnType().equals(Long.TYPE));
                }).collect(
                        Collectors.toMap(
                                n -> {
                                    String name =
                                            n.getName()
                                                    .replace("get", "")
                                                    .replace("JDBC", "Jdbc")
                                                    .replace("SQL", "Sql")
                                                    .replace("URL", "url");

                                    String formattedName = name.trim().substring(0, 1).toUpperCase() + name.substring(1);

                                    return formattedName;
                                },
                                m -> m
                        )
                );
    }
}

File: src/main/kotlin/org/dbshell/reflection/utils/DatabaseMetadataUtil.kt
package org.dbshell.reflection.utils

import java.lang.Boolean
import java.lang.reflect.Method
import java.lang.reflect.Modifier
import java.sql.DatabaseMetaData
import java.util.*
import java.util.stream.Collectors
import kotlin.String

object DatabaseMetadataUtil {
    fun getPrimitivesFromDBMetadata(dbmd: DatabaseMetaData): Map<String, Method?> {
        val klazz: Class<*> = dbmd.javaClass
        return Arrays.stream(klazz.methods)
            .filter { m: Method ->
                val modifier = m.modifiers
                m.parameterCount == 0 && Modifier.isPublic(modifier) &&
                (m.returnType == String::class.java || m.returnType == Integer.TYPE || m.returnType == Boolean.TYPE || m.returnType == java.lang.Long.TYPE)
            }.collect(
                Collectors.toMap(
                    { n: Method ->
                        val name = n.name
                            .replace("get", "")
                            .replace("JDBC", "Jdbc")
                            .replace("SQL", "Sql")
                            .replace("URL", "url")
                        val formattedName =
                            name.trim { it <= ' ' }.substring(0, 1).toUpperCase() + name.substring(1)
                        formattedName
                    },
                    { m: Method? -> m }
                )
            )
    }
}